include/miopen/config.h.in

```diff
@@ -78,11 +78,62 @@
 
 // clang-format off
 #cmakedefine MIOPEN_DEFAULT_FIND_MODE @MIOPEN_DEFAULT_FIND_MODE@
+
 // clang-format on
-#cmakedefine MIOPEN_AMDGCN_ASSEMBLER "@MIOPEN_AMDGCN_ASSEMBLER@"
-#cmakedefine HIP_OC_COMPILER "@HIP_OC_COMPILER@"
-#cmakedefine MIOPEN_HIP_COMPILER "@MIOPEN_HIP_COMPILER@"
-#cmakedefine MIOPEN_OFFLOADBUNDLER_BIN "@MIOPEN_OFFLOADBUNDLER_BIN@"
+#ifdef MIOPEN_COMPILER_FLAGS_V2 /* Flag to switch to MIOPEN Compiler Flag Approach V2*/
+  /* Include rocm-core header for get ROCm Install Path Method */
+  #include <rocm-core/rocm_getpath.h>
+
+  /* Include header for PATH_MAX */
+  #if defined(__linux__)
+  #include <linux/limits.h>
+  #endif /* __linux__ */
+  #if defined(_WIN32)
+  #include <windows.h>
+  #ifndef PATH_MAX
+  #define PATH_MAX MAX_PATH
+  #endif
+  #endif /* _WIN32*/
+
+  /* Max Buffer Len to hold the Flag Path */
+  #define PATH_MAX_LEN (PATH_MAX+1)
+
+  /* Flags to help identify/differentiate Definitions for each MIOPEN Compiler Flags */
+  typedef enum {
+      MIOPEN_AMDGCN_Flag=0,
+      MIOPEN_OC_COMPILER_Flag=1,
+      MIOPEN_HIP_COMPILER_Flag=2,
+      MIOPEN_OFFLOAD_BUNDLER_Flag=3,
+      MIOPEN_CMP_MAX_Flag=4 /*Should be last Flag definition*/
+  } MIOPEN_CMP_Flags_t;
+
+  /* Global variable prototype of List to hold the Compiler Flag Paths
+   * The definition of this variable expected to be available in C/C++
+   * code of the Target Library */
+  extern char MIOPEN_CMP_FLAGS[MIOPEN_CMP_MAX_Flag][PATH_MAX_LEN];
+
+  /* Function to generate the MIOPEN Compiler Flag Value using
+   * ROCm Base Install Path fetched using getROCmInstallPath()
+   * This approach depends on the getROCmInstallPath() provided by rocm-core
+   * This flag Paths are expected to be deprecated/modified in upcoming MAJOR Releases */
+  char *generateFlagValue(MIOPEN_CMP_Flags_t x);
+
+  /* MIOPEN Compiler Flags generated using ROCM Base install Path
+   * This will read the ROCM Base Install Path if it is already generated
+   * or (if required), will generate the MIOPEN Compiler Flag Path Values
+   * and return the value.*/
+  #cmakedefine GET_MIOPEN_CMP_FLAGS(X)   ((MIOPEN_CMP_FLAGS[X][0]!=0)?MIOPEN_CMP_FLAGS[X]:generateFlagValue(X))
```

Comment:
This is not thread-safe.

[Recommendation]
```C++
const char* GetPathToSomething() // you may wish to return std::string&
{
    static const std::string path = GetPathToSomethingImpl();
    return path.c_str();
}
```

Put prototypes of getters into this header and call them in MIOPEN_AMDGCN_ASSEMBLER and alike.

All other stuff can (and, therefore, should) be removed or moved to the .cpp file.

Reply:
Not yet resolved.

Reply:
@arvindcheru Please do not hesitate to contact me directly if something is unclear or is you need more detailed guidance. You can find my email in my github profile.

Reply:
Thanks for the comments, for this comment changes are done currently testing separately, the changes https://github.com/arvindcheru/MIOpen/pull/4,

Reply:
@arvindcheru Good. I've reviewed that changes, see https://github.com/arvindcheru/MIOpen/pull/4#pullrequestreview-1876376417. Please integrate those changes into this branch and we are ready for merging.

Reply:
integrated the changes from https://github.com/arvindcheru/MIOpen/pull/4